rr_mapper/src/laser_mapper.cpp

```diff
@@ -22,4 +24,447 @@ LaserMapper::LaserMapper(ros::NodeHandle nh) {
  * @return NONE
  */
 LaserMapper::~LaserMapper() {
+    belief_map_.clear();
 }
+
+/**
+ * @name PublishMap
+ * @brief Joins the entire map together
+ *        & Publishes the full_map
+ * @return NONE
+ */
+void LaserMapper::PublishMap() {
+    nav_msgs::OccupancyGrid full_map;
```

Comment:
I know we talked about this already but I'm not sure if you investigated this, but would it be a really difficult task to make this function work through synchronised policy instead of just storing the previous lidar and lane detection data in the class?

Reply:
Nope this not being changed cus lidar data might not return anything hence sync policy might screw stuff over